src/FlowPlayer.cpp

```diff
@@ -75,6 +75,11 @@ static void migrateCache()
         }
         QDir(QStandardPaths::writableLocation(QStandardPaths::GenericCacheLocation)).rmdir("flowplayer");
     }
+    // if the media-art directory does not exist, make it.
+    if (!QFileInfo(newCache).isDir()) {
+        QDir().mkpath( newCache );
+        QDir().mkpath( newCache + "/media-art" );
```

Comment:
I would have put this line in the main directly, ensuring that all necessary directories exist on startup. Besides, the `if exists` is superflous. The `mkpath` call is a no-op, as far as I know, if the directory already exists.

Reply:
Hmmm. It was a quick fix for the missing directory. As You've said, putting it in main makes more sense at this point.